gulpfile.babel.js

```diff
@@ -60,7 +61,31 @@ gulp.task('webpack', (cb) => {
   });
 });
 
-gulp.task('serve', () => {
+    
+ 
+// Setting up the test task 
+gulp.task('protractor', ['serve'], function(callback) {
+    gulp
+        .src(['example_spec.js'])
+        .pipe(gulpProtractorAngular({
+            'configFile': 'protractor.conf.js',
+            'debug': false,
+            'autoStartStopServer': true
+        }))
+        .on('error', function(e) {
+            console.log(e);
+        })
+        .on('end', callback);
+});
+
+
+gulp.task('e2e', ['protractor'], function() {
+  process.exit();
```

Comment:
Unsure what I'm missing here. Everything finishes except for the process.

`
// Setting up the test task 
gulp.task('protractor', ['serve'], function(callback) {
    gulp
        .src(['example_spec.js'])
        .pipe(gulpProtractorAngular({
            'configFile': 'protractor.conf.js',
            'debug': false,
            'autoStartStopServer': true
        }))
        .on('error', function(e) {
            console.log(e);
        })
        .on('end', callback);
});


gulp.task('e2e', ['protractor'], function(callback) {
  //process.exit();
  stopServer();
  callback();
});



gulp.task('serve', function(callback) {
  const config = require('./webpack.dev.config');
  config.entry.app = [
    // this modules required to make HRM working
    // it responsible for all this webpack magic
    'webpack-hot-middleware/client?reload=true',
    // application entry point
    paths.entry
  ];

  var compiler = webpack(config);

  const server = serve.init({
    port: process.env.PORT || 3000,
    open: false,
    server: {baseDir: root},
    middleware: [
      historyApiFallback(),
      webpackDevMiddelware(compiler, {
        stats: {
          colors: colorsSupported,
          chunks: false,
          modules: false
        },
        publicPath: config.output.publicPath
      }),
      webpachHotMiddelware(compiler)
    ]
  });

  stopServer = function(callback) {
    server.exit();
  }

  callback();
});
`